### ISSUE TITLE
Odd behaviour of 'mm day-1' units.

### ISSUE THREAD

**Issue by rcomer** [SciTools/cf-units#82]
I found some odd behaviour when manipulating units of mm day-1.  The `correlation_units` function below mimics what happens to the units of `iris` cubes with the same unit when they are passed to `iris.analysis.stats.pearsonr`.
```python
import cf_units
print cf_units.__version__

def correlation_units(unit_string):
    unit = cf_units.Unit(unit_string)
    unit_squared = unit * unit

    return unit_squared / unit_squared
    
for unit_string in ['K', 'm/s', 'mm day-1', 'm', 'hPa']:
    print unit_string, ': ', correlation_units(unit_string)
```

For most of the units, the function produces `cf_units.Unit('1')`, as expected, but for 'mm day-1', I get a strange unit of `cf_units.Unit('1 1')`.

```
1.1.3
K :  1
m/s :  1
mm day-1 :  1 1
m :  1
hPa :  1
```

**Comment by rcomer** [comment]
This strange new unit isn't quite what it looks like.  This statement evaluates to False:
```python
correlation_units('mm day-1') == cf_units.Unit('1 1')
```
However, if a cube with this unit is saved to NetCDF and reloaded, the equality returns True:
```python
import iris

test_cube = iris.cube.Cube(range(5), units=correlation_units('mm day-1'))
iris.save(test_cube, 'test.nc')
test_cube2 = iris.load_cube('test.nc')
test_cube2.units == cf_units.Unit('1 1')
```
This caused my code to break (at `iris.analysis.maths._assert_matching_units`) when trying to take a difference between a just-calculated correlation cube and one loaded from file.

I have an obvious work around for this, but thought I should flag it here too as the behaviour doesn't make sense to me.

**Comment by djkirkham** [comment]
Hi @rcomer 

This is very strange. It seems to be an issue with the underlying udunits library which cf_units uses. I've raised an issue with them (https://github.com/Unidata/UDUNITS-2/issues/52)

**Comment by pelson** [comment]
Apparently this is now fixed with udunits 2.2.26.
